packages/portfolio-contract/src/type-guards.ts

```diff
@@ -30,17 +30,27 @@ export type CompoundGive = {
   Compound: Amount<'nat'>;
 };
 export type GmpGive = {} | AaveGive | CompoundGive | (AaveGive & CompoundGive);
-export type OpenPortfolioGive = {
-  USDN?: Amount<'nat'>;
-  NobleFees?: Amount<'nat'>;
-  Access?: Amount<'nat'>;
-} & ({} | GmpGive);
+type USDNGive = { NobleFees?: Amount<'nat'> } & (
+  | { USDNSwapIn: Amount<'nat'> }
+  | { USDNLock: Amount<'nat'> }
+  | { USDNSwapOut: Amount<'nat'> }
+  | { USDNUnlock: Amount<'nat'> }
```

Comment:
swap-only and swap-and-lock are probably separate protocols that should be treated as separate positions.
IOU an issue.